fibonacci.js

```
@@ -0,0 +1,121 @@
+
+fibInput.addEventListener("focus", clearInput);
+
+buttonServer.addEventListener("click", getFibNumber);
+
+buttonLocal.addEventListener("click", () => {
+  spinner.classList.remove("d-none");
+  setTimeout(() => {
+    spinner.classList.add("d-none");
+    fibNumber.innerText = fibFunction(parseInt(fibInput.value));
+  }, 200);
+});
+
+// Swapping between local and server buttons
+checkBox.addEventListener("change", () => {
+  if (checkBox.checked) {
+    // hide the local button
+    buttonLocal.classList.add("d-none");
+    buttonServer.classList.remove("d-none");
+  } else if (!checkBox.checked) {
+    // hide the server button
+    buttonLocal.classList.remove("d-none");
+    buttonServer.classList.add("d-none");
+  }
+});
+
+// Function to calculate the Fibonacci number LOCALLY:
+function fibFunction(n) {
+  if (n < 0) {
+    error00.classList.remove("d-none");
+    fibNumber.classList.add("d-none");
+  } else if (n < 2) {
+    error00.classList.add("d-none");
+    return n;
+  } else if (n > 50) {
+    error00.classList.add("d-none");
+    error50.classList.remove("d-none");
+    fibNumber.classList.add("d-none");
+  } else {
+    error00.classList.add("d-none");
+    fibNumber.classList.remove("d-none");
+    return fibFunction(n - 1) + fibFunction(n - 2);
+  }
+}
+
+// Function to calculate the Fibonacci number ON THE SERVER:
+function getFibNumber() {
+  const index = fibInput.value;
+  if (index > 50) {
+    // if input is more than 50
+    spinner.classList.remove("d-none");
+    setTimeout(() => {
+      spinner.classList.add("d-none");
+      error50.classList.remove("d-none");
+    }, 500);
+  } else {
+    // if it is 50 or less
+    spinner.classList.remove("d-none");
+    fetch(`${"http://localhost:5050/fibonacci/"}${index}`)
+      .then((response) => {
+        if (!response.ok) {
+          response.text().then((error) => {
+            console.log("Server error:" + error);
+            fibNumber.innerText = `Server Error: ${error}`;
```

Instead of doing this here. You should throw the error and display it inside a .catch